x-pack/plugins/apm/public/services/rest/apm/apm.ts

```diff
@@ -19,3 +25,54 @@ export async function getEncodedEsQuery(kuery?: string) {
   const esFilterQuery = convertKueryToEsQuery(kuery, indexPattern);
   return encodeURIComponent(JSON.stringify(esFilterQuery));
 }
+
+export async function getKueryESFilter(kuery?: string): Promise<ESFilter | undefined> {
+  if (!kuery) {
+    return;
+  }
+
+  const indexPattern = await getAPMIndexPatternForKuery();
+  if (!indexPattern) {
+    return;
+  }
+
+  return convertKueryToEsQuery(kuery, indexPattern) as ESFilter;
+}
+
+export function getEnvironmentESFilter(environment?: string): ESFilter | undefined {
+
+  if (!environment) {
+    return undefined;
+  }
+
+  if (environment === 'Null') {
+    return {
+      bool: { must_not: { exists: { field: 'service.environment' } } }
+    };
+  }
+  return {
+    term: { 'service.environment': environment }
+  };
+}
+
+export function getUIFilters(urlParams: IUrlParams): UIFilters {
+  return {
+    global: {
+      kuery: urlParams.kuery,
+      environment: urlParams.environment
+    }
+  };
+}
+
+export async function getEncodedEsFilters(uiFilters: UIFilters): Promise<string> {
+  const { global } = uiFilters;
+  const kuery = await getKueryESFilter(global.kuery);
+  const environment = getEnvironmentESFilter(global.environment);
+
+  const filters = [
+    kuery,
+    environment,
+  ].filter(filter => !!filter) as ESFilter[];
+
+  return encodeURIComponent(JSON.stringify(filters));
+}
```

Comment:
What about creating a folder called `filters` that contains kuery.ts and environment.ts?

Reply:
I also think that `UIFilters` should be moved from `/typings` to the filters folder

Reply:
The "filters" folder should probably be in the "UrlParamsContext" folder

Reply:
great point, in line 7 i left a note to myself to rename this file, but i think it can use more than just a rename for code org.